src/repos/dfusion/DfusionRepoImpl.ts

```diff
@@ -0,0 +1,57 @@
+import Web3 from 'web3'
+
+import Logger from '../../helpers/Logger'
+import { DfusionRepo, WatchOrderPlacementParams } from 'types'
+import { StablecoinConverter } from '../../contracts/StablecoinConverter'
+
+const log = new Logger('repos:dfusion')
+
+interface Params {
+  stableCoinConverterContract: StablecoinConverter
+  web3: Web3
+}
+
+export class DfusionRepoImpl implements DfusionRepo {
+  private _web3: Web3
+  private _contract: StablecoinConverter
+
+  constructor (params: Params) {
+    const { web3, stableCoinConverterContract } = params
+    log.debug('Setup dfusionRepo with contract address %s', stableCoinConverterContract.options.address)
+
+    this._contract = stableCoinConverterContract
+    this._web3 = web3
+  }
+
+  public watchOrderPlacement (params: WatchOrderPlacementParams) {
+    this._contract.events
+      .OrderPlacement({}, error => params.onError(error))
```

Comment:
callback is reduntant if you already have `on 'data' and on 'error'`
do `.OrderPlacement()`

Reply:
I thought so! But then, there was an error that I was not being notified of, until I've added that callback. This is why I'm so happy about web3 lately

Try to remove that, and do not provide the contract address, you'll see what I mean

Reply:
Well, that's unfortunate. Maybe post an issue in their repo?